src/status_im/utils/ethereum/erc20.cljs

```diff
@@ -57,4 +62,128 @@
 (defn allowance [web3 contract owner-address spender-address cb]
   (ethereum/call web3
                  (ethereum/call-params contract "allowance(address,address)" (ethereum/normalized-address owner-address) (ethereum/normalized-address spender-address))
-                 #(cb %1 (ethereum/hex->bignumber %2))))
\ No newline at end of file
+                 #(cb %1 (ethereum/hex->bignumber %2))))
+
+(defn- parse-json [s]
+  (try
+    (let [res (-> s
+                  js/JSON.parse
+                  (js->clj :keywordize-keys true))]
+      (if (and (:error res)
+               (= (:error res) ""))
+        {:result true}
+        res))
+    (catch :default e
+      {:error (.-message e)})))
+
+(defn- add-padding [address]
+  (when address
+    (str "0x000000000000000000000000" (subs address 2))))
+
+(defn- remove-padding [topic]
+  (if topic
+    (str "0x" (subs topic 26))))
+
+(defn- parse-transaction-entry [block-number chain direction entries]
+  (into {}
+        (for [entry entries]
+          [(:transactionHash entry)
+           {:block         (-> entry :blockNumber ethereum/hex->int str)
+            :hash          (:transactionHash entry)
+            :symbol        (->> entry :address (tokens/address->token chain) :symbol)
+            :from          (-> entry :topics second remove-padding)
+            :to            (-> entry :topics last remove-padding)
+            :value         (-> entry :data ethereum/hex->bignumber)
+            :type          direction
+
+            :confirmations (str (- block-number (-> entry :blockNumber ethereum/hex->int)))
+
+            :gas-price     "0"
+            :nonce         "0"
+            :data          "0x"
+
+            :gas-limit     "0"
+            ;; NOTE(goranjovic) - timestamp is mocked to the current time so that the transaction is shown at the
+            ;; top of transaction history list between the moment when transfer event was detected and actual
+            ;; timestamp was retrieved from block info.
+            :timestamp     (str (.getTime (cljs-time/now)))
```

Comment:
Isn't there a clojure equivalent to getTime? Also tis should be a utilitu fn.

Reply:
I'm pretty sure we had `now-ms` before somewhere around, but don't see it anywhere now. Maybe I should just add it.